platform-common-string/src/test/java/com/softicar/platform/common/string/csv/CsvTokenizerTest.java

```diff
@@ -0,0 +1,285 @@
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class CsvTokenizerTest extends Assert {
+
+	private List<List<String>> tokenMatrix;
+
+	public CsvTokenizerTest() {
+
+		this.tokenMatrix = null;
+	}
+
+	// ---------------- single line ---------------- //
+
+	@Test
+	public void testWithSingleLine() {
+
+		tokenize("qwe,asd,zxc");
+		new Asserter()//
+			.assertRow("qwe", "asd", "zxc")
+			.assertNoMoreRows();
+	}
+
+	// ---------------- multiples lines ---------------- //
+
+	@Test
+	public void testWithMultipleLinesSeparatedByBackslashN() {
+
+		tokenize("qwe,asd,zxc\n123,456,789");
+		new Asserter()//
+			.assertRow("qwe", "asd", "zxc")
+			.assertRow("123", "456", "789")
+			.assertNoMoreRows();
+	}
+
+	@Test
+	public void testWithMultipleLinesSeparatedByBackslashR() {
+
+		tokenize("qwe,asd,zxc\r123,456,789");
+		new Asserter()//
+			.assertRow("qwe", "asd", "zxc")
+			.assertRow("123", "456", "789")
+			.assertNoMoreRows();
+	}
+
+	@Test
+	public void testWithMultipleLinesSeparatedByBackslashRN() {
+
+		tokenize("qwe,asd,zxc\r\n123,456,789");
+		new Asserter()//
+			.assertRow("qwe", "asd", "zxc")
+			.assertRow("123", "456", "789")
+			.assertNoMoreRows();
+	}
+
+	@Test
+	public void testWithMultipleLinesAndEmptyLine() {
+
+		tokenize("qwe,asd,zxc\n\n123,456,789");
+		new Asserter()//
+			.assertRow("qwe", "asd", "zxc")
+			.assertRow("123", "456", "789")
+			.assertNoMoreRows();
+	}
+
+	@Test
+	public void testWithMultipleLinesOfVariableLength() {
+
+		tokenize("qwe,asd,zxc\n123,456\nxxx,yyy,zzz,___");
+		new Asserter()//
+			.assertRow("qwe", "asd", "zxc")
+			.assertRow("123", "456")
+			.assertRow("xxx", "yyy", "zzz", "___")
+			.assertNoMoreRows();
+	}
+
+	// ---------------- value separation ---------------- //
+
+	@Test
+	public void testWithValueSeparationByAdjacentCommas() {
+
+		tokenize(",,qwe,asd,, ,zxc,,");
+		new Asserter()//
+			.assertRow("", "", "qwe", "asd", "", " ", "zxc", "", "")
+			.assertNoMoreRows();
+	}
+
+	@Test
+	public void testWithValueSeparationMissing() {
+
+		try {
+			tokenize("\"qwe\" \"asd\",zxc");
+		} catch (CsvFormatException exception) {
+			assertEquals(
+				"Invalid CSV format after character 5: After a double quoted value, a comma, a newline or end-of-file are expected.",
+				exception.getMessage());
+			return;
+		}
+		fail();
+	}
+
+	// ---------------- line-spanning value ---------------- //
+
+	@Test
+	public void testWithLineSpanningValue() {
+
+		tokenize("\"X\nX\r\nX\rX\",asd,zxc");
+		new Asserter()//
+			.assertRow("X\nX\r\nX\rX", "asd", "zxc")
+			.assertNoMoreRows();
+	}
+
+	// ---------------- spacing ---------------- //
+
+	@Test
+	public void testWithSpacesAroundValues() {
+
+		tokenize(" qwe, asd ,zxc ");
+		new Asserter()//
+			.assertRow(" qwe", " asd ", "zxc ")
+			.assertNoMoreRows();
+	}
+
+	// ---------------- quoting ---------------- //
+
+	@Test
+	public void testWithQuotes() {
+
+		tokenize("\"\"\"qwe,123\"\"\",\"asd\",zxc");
```

Comment:
replace with:
`tokenize("\"\"\"qwe,123\"\"\",asd,zxc");`

Reply:
This test method is _supposed_ to test both cases (enclosing in single-quotes and in triple quotes).
If you insist, we could split the method in two.